sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala

```diff
@@ -63,7 +63,14 @@ class QueryExecution(
   // TODO: Move the planner an optimizer into here from SessionState.
   protected def planner = sparkSession.sessionState.planner
 
-  def assertAnalyzed(): Unit = analyzed
+  def assertAnalyzed(): Unit = {
+    try {
+      analyzed
+    } catch {
+      case e: AnalysisException =>
+        SQLExecution.withNewExecutionId(this, Some("analyze"))(throw e)
```

Comment:
is this change necessary?

Reply:
This is the key change of this PR to capture failures from analysis

Reply:
shouldn't `SQLExecution` be the only place to try-catch the errors? including analysis errors.

Reply:
oh I see what you mean, for simple `df.select(...)`, no `SQLExecution` is created. We need to specifically create an executio here.

Reply:
The reason is we do eager analysis for dataframe. Can we add some comments to explain it?

Reply:
updated